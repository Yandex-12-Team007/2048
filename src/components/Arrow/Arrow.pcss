.arrow {
  cursor: pointer;
  height: 12px;
  width: 8px;
  position: relative;

  .arrow-top,
  .arrow-bottom {
    background: var(--gray);
    height: 1px;
    left: -3px;
    position: absolute;
    top: 6px;
    width: 10px;
  }

  .arrow-top::after,
  .arrow-bottom::after {
    background: var(--orange);
    content: "";
    height: 100%;
    position: absolute;
    top: 0;
    transition: all 0.15s;
  }

  .arrow-top {
    top: 6px;
    transform: rotate(45deg);
    transform-origin: bottom right;

    &:after {
      left: 100%;
      right: 0;
      transition-delay: 0s;
    }
  }

  .arrow-bottom {
    transform: rotate(-45deg);
    transform-origin: top right;

    &:after {
      left: 0;
      right: 100%;
      transition-delay: 0.15s;
    }
  }

  &:hover .arrow-top::after {
    left: 0;
    transition-delay: 0.15s;
  }

  &:hover .arrow-bottom::after {
    right: 0;
    transition-delay: 0s;
  }
}

.arrow-left {
  transform: rotate(0);
}

.arrow-right {
  transform: rotate(180deg);
}

.arrow-bottom {
  transform: rotate(270deg);
}

.arrow-top {
  transform: rotate(90deg);
}
