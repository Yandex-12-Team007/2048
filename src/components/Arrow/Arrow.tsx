import React from 'react';

import './Arrow.pcss';

export enum ArrowDirection {
  LEFT = 'left',
  RIGHT = 'right',
  TOP = 'bottom',
  BOTTOM = 'top'
}

export default function Arrow({direction = 'left'}) {
  return <div className={'arrow arrow-'+direction}>
    <div className="arrow-top" />
    <div className="arrow-bottom" />
  </div>
}
