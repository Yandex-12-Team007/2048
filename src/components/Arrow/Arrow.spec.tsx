import React from 'react';
import renderer from 'react-test-renderer';
import Arrow, {ArrowDirection} from './Arrow';

it('Arrow top', () => {
  const tree = renderer.create(
      <Arrow direction={ArrowDirection.TOP}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Arrow bottom', () => {
  const tree = renderer.create(
      <Arrow direction={ArrowDirection.TOP}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Arrow left', () => {
  const tree = renderer.create(
      <Arrow direction={ArrowDirection.LEFT}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Arrow right', () => {
  const tree = renderer.create(
      <Arrow direction={ArrowDirection.RIGHT}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
