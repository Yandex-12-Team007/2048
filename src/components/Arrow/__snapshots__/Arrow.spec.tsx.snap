// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrow bottom 1`] = `
<div
  className="arrow arrow-bottom"
>
  <div
    className="arrow__top"
  />
  <div
    className="arrow__bottom"
  />
</div>
`;

exports[`Arrow left 1`] = `
<div
  className="arrow arrow-left"
>
  <div
    className="arrow__top"
  />
  <div
    className="arrow__bottom"
  />
</div>
`;

exports[`Arrow right 1`] = `
<div
  className="arrow arrow-right"
>
  <div
    className="arrow__top"
  />
  <div
    className="arrow__bottom"
  />
</div>
`;

exports[`Arrow top 1`] = `
<div
  className="arrow arrow-bottom"
>
  <div
    className="arrow__top"
  />
  <div
    className="arrow__bottom"
  />
</div>
`;
